drivers/i2c/i2c_litex.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2019 Antmicro <www.antmicro.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include "i2c_bitbang.h"
+
+#define SCL_BIT_POS                0
+#define SDA_DIR_BIT_POS            1
+#define SDA_BIT_W_POS              2
+#define SDA_BIT_R_POS              0
+
+#define SDA_DIR_OUTPUT             1
+#define SDA_DIR_INPUT              0
+
+#define HIGH_STATE_ON_I2C_LINES    0x7
+
+struct i2c_litex_cfg {
+	struct i2c_bitbang bitbang;
+	volatile u32_t *w_reg;
+	volatile u32_t *r_reg;
+};
+
+#define GET_I2C_CFG(dev)						     \
+	((struct i2c_litex_cfg *) dev->config->config_info)
+
+static inline void set_bit(volatile u32_t *reg, u32_t bit, u32_t val)
+{
+	u32_t mask = (1 << bit);
```

Comment:
there should be an empty line here

Reply:
Also use `BIT(bit)` instead of shifting a signed integer, which produces undefined behavior `bit` is 31.

Please do that throughout the driver.

Reply:
Fixed